src/overlays/actors/ovl_En_Zf/z_en_zf.c

```diff
@@ -452,9 +452,15 @@ s16 EnZf_FindNextPlatformAwayFromPlayer(Vec3f* pos, s16 curPlatform, s16 arg2, G
         }
     }
 
-    // These functions have no side effects, so these two calls do nothing
+    //! @bug `altNextPlatform` can be -1 in certain conditions and cause an out of bounds access.
+    //! Under normal conditions, this doesn't cause problems because the data before `sPlatformPositions`
+    //! is section padding between .text and .data, so 0 gets read as a float.
+#ifndef AVOID_UB
+    // Defining `AVOID_UB` will remove these function calls entirely as they don't 
+    // have any effect (and return values are not used).
     Math_Vec3f_DistXYZ(&player->actor.world.pos, &sPlatformPositions[nextPlatform]);
     Math_Vec3f_DistXYZ(&player->actor.world.pos, &sPlatformPositions[altNextPlatform]);
+#endif
```

Comment:
Agreed on vanilla code going first.
Not sure on an else just for the comment though. I think the way it is now its fine, if you go through the trouble of removing all AVOID_UB blocks, you dont care about it anyway and dont need the explanation of it

Reply:
To be clear if one was to process the source to remove avoid_ub blocks, hypothetically
```c
#ifndef AVOID_UB
    // Defining `AVOID_UB` will remove these function calls entirely as they don't 
    // have any effect (and return values are not used).
    Math_Vec3f_DistXYZ(&player->actor.world.pos, &sPlatformPositions[nextPlatform]);
    Math_Vec3f_DistXYZ(&player->actor.world.pos, &sPlatformPositions[altNextPlatform]);
#endif
```
would become
```c
    // Defining `AVOID_UB` will remove these function calls entirely as they don't 
    // have any effect (and return values are not used).
    Math_Vec3f_DistXYZ(&player->actor.world.pos, &sPlatformPositions[nextPlatform]);
    Math_Vec3f_DistXYZ(&player->actor.world.pos, &sPlatformPositions[altNextPlatform]);
```
and indeed since in this scenario you wouldn't care about the avoid_ub code you would want the comments getting rid of, hence why I suggested instead having
```c
#ifndef AVOID_UB
    Math_Vec3f_DistXYZ(&player->actor.world.pos, &sPlatformPositions[nextPlatform]);
    Math_Vec3f_DistXYZ(&player->actor.world.pos, &sPlatformPositions[altNextPlatform]);
#else
    // Defining `AVOID_UB` will remove these function calls entirely as they don't 
    // have any effect (and return values are not used).
#endif
```
which would become
```c
    Math_Vec3f_DistXYZ(&player->actor.world.pos, &sPlatformPositions[nextPlatform]);
    Math_Vec3f_DistXYZ(&player->actor.world.pos, &sPlatformPositions[altNextPlatform]);
```
without the avoid_ub comments

in this case it's a bit weird because the avoid_ub block is empty code-wise, but that's also exactly the "fix" (having no code) so it kinda makes sense imo

it also clearly separate avoid_ub comments from the vanilla code, keeping the vanilla code sticked to the `@bug`

Reply:
oh yea I see now, thanks. will change